### PR TITLE
remove deprecated metric and promote the replacement to STABLE

### PR DESCRIPTION
**Pull request by logicalhan**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This PR removes the deprecated `apiserver_longrunning_gauge` metric and promotes its replacement `apiserver_longrunning_requests`.

#### Does this PR introduce a user-facing change?

```release-note
apiserver_longrunning_gauge is removed from the codebase. Please use apiserver_longrunning_requests instead. 
```

### CODE REVIEW

**Comment by logicalhan** [comment]
/hold

(I'm an approver so I want to ensure this gets vetted properly in reviews).

**Review by dgrisonnet — APPROVED** [review]
Since the deprecated metric and the alpha one replacing it have the same labels, I think it is fine to graduate the new metric to stable at the same time as removing the old one since users have been using this API for a while now without needing to extend it.

For what it's worth, I would have been against doing the two changes during the same release if the labels were different. That's because unless the metric is completely removed, you can't make sure that users transitioned to the new API and have experienced it.

/lgtm
/approve

**Comment by logicalhan** [comment]
> Since the deprecated metric and the alpha one replacing it have the same labels, I think it is fine to graduate the new metric to stable at the same time as removing the old one since users have been using this API for a while now without needing to extend it.
> 
> For what it's worth, I would have been against doing the two changes during the same release if the labels were different. That's because unless the metric is completely removed, you can't make sure that users transitioned to the new API and have experienced it.

It was actually already deprecated in a previous release.

**Pull request merged**
